File: Professional/ActiveReportsWithMVC/VB.NET/ActiveReportsWithMVC/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.Models
Namespace Controllers
	Public Class HomeController
		Inherits Controller
		''' <summary>
		''' Returns view Index.cshtml
		''' Passes the Entire Customers list to its model
		''' </summary>
		''' <returns>Index.cshtml</returns>
		Public Function Index() As ActionResult
			Return View("Index", Repository.GetCustomers())
		End Function
		''' <summary>
		''' Start method
		''' </summary>
		''' <returns></returns>
		Public Function Start() As ActionResult
			Return Index()
		End Function
		''' <summary>
		''' Returns View Details.cshtml
		''' </summary>
		''' <param name="id">Customer ID</param>
		''' <returns>Details.cshtml</returns>
		Public Function Details(ByVal id As String) As ActionResult
			Return View("Details", Repository.GetCustomer(id))
		End Function
		''' <summary>
		''' Returns View Viewer.cshtml
		''' </summary>
		''' <param name="id">Customer id</param>
		''' <param name="reporttype">Section/Page</param>
		''' <returns>Viewer.cshtml</returns>
		Public Function ViewReport(ByVal id As String, ByVal reporttype As String) As ActionResult
			Return View("Viewer", New ReportDescriptor() With { _
			 .Id = id, _
			 .Type = reporttype _
			})
		End Function
	End Class
End Namespace

File: Professional/ActiveReportsWithMVC/C#/ActiveReportsWithMVC/Controllers/HomeController.cs
using System.Web.Mvc;
using GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.Models;
namespace GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.Controllers
{
	public class HomeController : Controller
	{
		/// <summary>
		/// Returns view Index.cshtml
		/// Passes the Entire Customers list to its model
		/// </summary>
		/// <returns>Index.cshtml</returns>
		public ActionResult Index()
		{
			
			return View("Index", Repository.GetCustomers());
		 
			
		}
		/// <summary>
		/// Start method
		/// </summary>
		/// <returns></returns>
		public ActionResult Start()
		{
			return Index();
		}
		/// <summary>
		/// Returns View Details.cshtml
		/// </summary>
		/// <param name="id">Customer ID</param>
		/// <returns>Details.cshtml</returns>
		public ActionResult Details(string id)
		{
			return View("Details",Repository.GetCustomer(id) );
		}
		/// <summary>
		/// Returns View Viewer.cshtml
		/// </summary>
		/// <param name="id">Customer id</param>
		/// <param name="reporttype">Section/Page</param>
		/// <returns>Viewer.cshtml</returns>
		public ActionResult ViewReport(string id,string reporttype)
		{
			return View("Viewer", new ReportDescriptor { Id = id, Type = reporttype });
		  
		}
	}
}
